### PR TITLE
Use MMEncoderAttention (=use FlashAttention) instead of torch.sdpa in radio.py

### PR DESCRIPTION
**Pull request by netanel-haber**
Use attention masking mechanism via MMEncoderAttention (=use FlashAttention when possible) instead torch.sdpa in radio.py

DocVQA_VAL and InfoVQA_VAL equivalent before and after.

### CODE REVIEW

File: vllm/model_executor/models/radio.py [hunk 1]
```diff
@@ -590,44 +590,36 @@ def _init_img_size(self, patch_size, img_size: int | tuple[int, int]):
...
         seq_lens = calc_seq_lens(imgs_sizes, patch_size)
-        patch_counts = [seq_len + num_skip for seq_len in seq_lens]
-        total_patches = sum(patch_counts)
-
-        # Create attention mask - default to False (mask out)
-        mask = torch.zeros(
-            total_patches, total_patches, dtype=torch.bool, device=device
+        adjusted = [s + num_skip for s in seq_lens]
+        cu_seqlens = torch.tensor(
+            list(accumulate(adjusted, initial=0)), dtype=torch.int32, device=device
```
Comment (nvnbagrov) [inline vllm/model_executor/models/radio.py:602]: Would be more readable to start with `seq_lens` as a tensor:
```python
seq_lens = torch.tensor(seq_lens, dtype=torch.int32, device=device)
seq_lens += num_skip  # or with the line avobe
cu_seqlens = seq_lens.cumsum(dim=0)
```

Comment (nvnbagrov) [reply vllm/model_executor/models/radio.py:602]: Below would be `seq_lens.max().cpu()`

Comment (netanel-haber) [reply vllm/model_executor/models/radio.py:602]: I think my current code is probably more performant, but I don't think its time-effective to benchmark it. If the vllm reviewer really wants me to change to your suggestion (or close to it), I'll do so. Leaving unresolved for now so they can see it

Comment (nvnbagrov) [reply vllm/model_executor/models/radio.py:602]: Indeed faster :) about 2x.
(For lengths up to ~100, reasonable assumption)

Comment (netanel-haber) [reply vllm/model_executor/models/radio.py:602]: Micro benchmarks are so much fun

**Review by ywang96 — APPROVED** [review]
